tests/MongoDB.Driver.Tests/AggregateFluentTests.cs

```diff
@@ -193,6 +194,297 @@ public void ChangeStream_should_add_the_expected_stage_when_options_is_null()
+
+            var ordersCollection = client.GetDatabase(databaseName).GetCollection<Order>(ordersCollectionName);
+            var warehousesCollection = client.GetDatabase(databaseName).GetCollection<Warehouse>(warehousesCollectionName);
+
+            var orderDocuments = new[]
+            {
+                new Order { Item = "almonds", Price = 12, Ordered = 2 },
+                new Order { Item = "pecans", Price = 20, Ordered = 1 },
+                new Order { Item = "cookies", Price = 10, Ordered = 60 }
+            };
+            ordersCollection.InsertMany(orderDocuments);
+
+            var warehouseDocuments = new[]
+            {
+                new Warehouse { StockItem = "almonds", Instock = 120 },
+                new Warehouse { StockItem = "pecans", Instock = 80 },
+                new Warehouse { StockItem = "almonds", Instock = 60 },
+                new Warehouse { StockItem = "cookies", Instock = 40 },
+                new Warehouse { StockItem = "cookies", Instock = 80 },
+            };
+            warehousesCollection.InsertMany(warehouseDocuments);
+
+            var pipeline = new EmptyPipelineDefinition<Warehouse>()
+                .Match(new BsonDocument("$expr",
+                    new BsonDocument("$and", new BsonArray
+                    {
+                        new BsonDocument("$eq", new BsonArray { "$stock_item", "$$order_item" }),
+                        new BsonDocument("$gte", new BsonArray { "$instock", "$$order_qty" })
+                    })))
+                .Project<Warehouse, Warehouse, StockData>(Builders<Warehouse>
+                    .Projection
+                        .Exclude(warehouses => warehouses.StockItem)
+                        .Exclude(warehouses => warehouses.Id));
+
+            var result = ordersCollection
+                .Aggregate()
+                .Lookup(warehousesCollectionName,
+                    new BsonDocument { { "order_item", "$item" }, { "order_qty", "$ordered" } },
+                    pipeline,
+                    new ExpressionFieldDefinition<Order, IEnumerable<StockData>>(order => order.StockData))
+                .ToList()
+                .Select(item =>
+                {
+                    var document = item.ToBsonDocument();
+                    document.Remove("_id");
+                    return document;
+                })
+                .ToList();
+
+            result.Count.Should().Be(3);
+            result[0].Should().Be("{ 'item' : 'almonds', 'price' : 12, 'ordered' : 2, 'stockdata' : [{ 'instock' : 120 }, { 'instock' : 60 }] }");
+            result[1].Should().Be("{ 'item' : 'pecans', 'price' : 20, 'ordered' : 1, 'stockdata' : [{ 'instock' : 80 }] }");
+            result[2].Should().Be("{ 'item' : 'cookies', 'price' : 10, 'ordered' : 60, 'stockdata' : [{ 'instock' : 80 }] }");
+        }
+
+        [Fact]
+        public void Lookup_with_let_and_mismatched_pipeline_condition_should_return_the_expected_result()
```

Comment:
I don't understand exactly what you are testing in this method or what "mismatched pipeline" means.

Reply:
I wanted to check how this lookup functionality will look like in case if the match condition is false for all fields combinations from two collections. Maybe this test is redundant.

Reply:
This does seem like a worthwhile test because you never know whether the server does something weird when the pipeline returns zero document.